src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-revisions-controller.php

```diff
@@ -312,6 +312,15 @@ public function prepare_item_for_response( $post, $request ) {
 		$theme_json = null;
 
 		if ( ! empty( $global_styles_config['styles'] ) || ! empty( $global_styles_config['settings'] ) ) {
+			/*
+			 * Register block style variations from the revision data.
```

Comment:
One small nit, I believe it's the theme.json files that the block style variations here are being registered from, rather than the revision data.

```suggestion
			 * Register block style variations from the theme data.